### ISSUE TITLE
GradCAM.py 中module.register_backward_hook 无输出

### ISSUE THREAD

**Issue by julyxia1992** [yizt/Grad-CAM.pytorch#15]
我在加载自己的模型后，调用
_get_grads_hook(self, module, input_grad, output_grad):
却没有输出？

**Comment by yizt** [comment]
@julyxia1992 您好，请问是怎么确定_get_grads_hook没有输出的？_get_grads_hook无需显示调用，loss反向传播会自动调用

**Comment by julyxia1992** [comment]
非常感谢作者的代码！本人终于可以实现目标检测的热力图

**Comment by DamonAtSjtu** [comment]
When I load my own model ,I also met this question, my model cannot call _get_grads_hook function automatically, how do you solve this question, thanks.

**Comment by julyxia1992** [comment]
> When I load my own model ,I also met this question, my model cannot call _get_grads_hook function automatically, how do you solve this question, thanks.

You must set your image like this : input.requires_grad_(True),

**Comment by julyxia1992** [comment]
> When I load my own model ,I also met this question, my model cannot call _get_grads_hook function automatically, how do you solve this question, thanks.

    image = torch.as_tensor(image.astype("float32").transpose(2, 0, 1)).requires_grad_(True)

**Comment by withtimesgo1115** [comment]
> > When I load my own model ,I also met this question, my model cannot call _get_grads_hook function automatically, how do you solve this question, thanks.
> 
> ```
> image = torch.as_tensor(image.astype("float32").transpose(2, 0, 1)).requires_grad_(True)
> ```
> 
> this line is in demo_retinanet.py lines140
> if you did not add requires_grad_(True),it cannot call _get_grads_hook function

请问我求出来的gradient全是0如何解决啊？

**Comment by Fly-dream12** [comment]
I have made the requires_grad_ as True, but i still can not get the gradient. Can you give some suggestions? @julyxia1992
